lib/python/pyflyby/_comms.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import absolute_import, division
+
+from   pyflyby._log             import logger
+import six
+
+# These are comm targets that the frontend (lab/notebook) is expected to
+# register. At this point, we handle only missing imports and
+# formatting imports
+
+MISSING_IMPORTS = "pyflyby.missing_imports"
+FORMATTING_IMPORTS = "pyflyby.format_imports"
+
+pyflyby_comm_targets= [MISSING_IMPORTS, FORMATTING_IMPORTS]
+
+# A map of the comms opened with a given target name.
+comms = {}
+
+# TODO: Document the expected contract for the different
+# custom comm messages
+
+def _register_target(target_name):
+    try:
+        ip = get_ipython()
+    except:
+        logger.debug("get_ipython() doesn't work")
+    else:
+        comm_manager = ip.kernel.comm_manager
+        comm_manager.register_target(target_name, comm_open_handler)
+
+
+def initialize_comms():
+    for target in pyflyby_comm_targets:
```

Comment:
don't do anything when using ipython CLI

Reply:
This has been changed - comm messages will be attempted only if a Jupyter environment is detected (like notebook, lab, jupyter console)